### DISCUSSION TITLE
How can I alter / mock IResolverContext in a unit test?

### DISCUSSION THREAD

**Discussion by ghost** [ChilliCream/hotchocolate#4711] [Q&A] [answered]
I'm using hot chocolate for my GraphQL APIs and am wondering how I go about altering the `IResolverContext` in my unit tests.

I start by creating a service collection containing the GraphQL setup.

```c#
services
    .AddGraphQLServer()
    .AddAuthorization()
    .AddQueryType<Query>()
    .AddMutationType<Mutation>()
    .AddTypes(types)
    .AddFiltering()
    .AddSorting()
    .AddMaxExecutionDepthRule(20)
    .AddType(new UuidType('N'));
```

I then get the request executor

```c#
var requestExecutor = await ServiceProvider.GetRequiredService<IRequestExecutorResolver>().GetRequestExecutorAsync();
```

Followed by executing the test using a Graph QL file.

```c#
var request = CreateQueryRequest("GetApplicationPermissionsAsync_Success.graphql");
var result = await requestExecutor.ExecuteAsync(request);
```

What I'm trying to achieve is a way of altering / mocking the request context so I can have unit tests that verify my authorization policies, I'm using NUnit and Moq with Hot Chocolate 12.6.0. 

Thank you for taking the time to take a look, really appreciate it.

**Comment by michaelstaib** [comment, marked as answer]
So in the case of auth, you can just pass in with your request a claims principal that is configured to your auth requirements.

In order to do that you the QueryRequestBuilder and add a property with the key nameof(ClaimsPrincipal) and as the value the ClaimsPrincipal

**Reply by ghost** [reply]
That worked perfectly, thank you.
